pkg/operator/ceph/object/bucket/s3-handlers.go

```diff
@@ -0,0 +1,243 @@
+	GetBucketPolicy                action = "s3:GetBucketPolicy"
+	GetBucketRequestPayment        action = "s3:GetBucketRequestPayment"
+	GetBucketTagging               action = "s3:GetBucketTagging"
+	GetBucketVersioning            action = "s3:GetBucketVersioning"
+	GetBucketWebsite               action = "s3:GetBucketWebsite"
+	GetLifecycleConfiguration      action = "s3:GetLifecycleConfiguration"
+	GetObjectAcl                   action = "s3:GetObjectAcl"
+	GetObject                      action = "s3:GetObject"
+	GetObjectTorrent               action = "s3:GetObjectTorrent"
+	GetObjectVersionAcl            action = "s3:GetObjectVersionAcl"
+	GetObjectVersion               action = "s3:GetObjectVersion"
+	GetObjectVersionTorrent        action = "s3:GetObjectVersionTorrent"
+	GetReplicationConfiguration    action = "s3:GetReplicationConfiguration"
+	ListAllMyBuckets               action = "s3:ListAllMyBuckets"
+	ListBucketMultiPartUploads     action = "s3:ListBucketMultiPartUploads"
+	ListBucket                     action = "s3:ListBucket"
+	ListBucketVersions             action = "s3:ListBucketVersions"
+	ListMultipartUploadParts       action = "s3:ListMultipartUploadParts"
+	PutAccelerateConfiguration     action = "s3:PutAccelerateConfiguration"
+	PutBucketAcl                   action = "s3:PutBucketAcl"
+	PutBucketCORS                  action = "s3:PutBucketCORS"
+	PutBucketLogging               action = "s3:PutBucketLogging"
+	PutBucketNotification          action = "s3:PutBucketNotification"
+	PutBucketPolicy                action = "s3:PutBucketPolicy"
+	PutBucketRequestPayment        action = "s3:PutBucketRequestPayment"
+	PutBucketTagging               action = "s3:PutBucketTagging"
+	PutBucketVersioning            action = "s3:PutBucketVersioning"
+	PutBucketWebsite               action = "s3:PutBucketWebsite"
+	PutLifecycleConfiguration      action = "s3:PutLifecycleConfiguration"
+	PutObjectAcl                   action = "s3:PutObjectAcl"
+	PutObject                      action = "s3:PutObject"
+	PutObjectVersionAcl            action = "s3:PutObjectVersionAcl"
+	PutReplicationConfiguration    action = "s3:PutReplicationConfiguration"
+	RestoreObject                  action = "s3:RestoreObject"
+)
+
+const version = "2012-10-17"
+
+type effect string
+
+const (
+	effectAllow effect = "Allow"
+	effectDeny  effect = "Deny"
+)
+
+type PolicyStatement struct {
+	Sid       string              `json:"Sid"`
+	Effect    effect              `json:"Effect"`
+	Principal map[string][]string `json:"Principal"`
+	Action    []action            `json:"Action"`
+	Resource  []string            `json:"Resource"`
+}
+
+type BucketPolicy struct {
+	Id        string            `json:"Id"`
+	Version   string            `json:"Version"`
+	Statement []PolicyStatement `json:"Statement"`
+}
+
+func NewBucketPolicy() *BucketPolicy {
+	bp := &BucketPolicy{
+		Version:   version,
+		Statement: []PolicyStatement{},
+	}
+	return bp
```

Comment:
any value in the assignment to `bp` vs just returning the struct?

Reply:
no, must not have been firing on all pistons, will change.

Reply:
no worries and it's very much a nit that most optimizers would handle...